src/Servers/Kestrel/perf/Microbenchmarks/Http3/Http3ConnectionBenchmarkBase.cs

```diff
@@ -72,6 +72,8 @@ public async Task MakeRequest()
 
         var stream = await _http3.CreateRequestStream(_requestHeadersEnumerator, _headerHandler);
 
+        _requestHeadersEnumerator.Initialize(_httpRequestHeaders);
+
         await stream.SendHeadersAsync(_requestHeadersEnumerator);
```

Comment:
Doesn't `_http3.CreateRequestStream` already call `SendHeadersAsync`? I wonder why this test was calling it twice. We probably shouldn't re-enumerate it though.

https://github.com/dotnet/aspnetcore/blob/600eb9aa53c052ec7327e2399744215dbe493a89/src/Servers/Kestrel/shared/test/Http3/Http3InMemory.cs#L445

Maybe we should just delete this line? Do we get a response if we don't call `SendHeadersAsync` again?

Reply:
With it removed, we now get a single 45-byte frame, rather than an immediate null (i.e. I'm pretty sure it was broken).